appinfo.go

```diff
@@ -20,9 +30,55 @@ type AppengineInfo interface {
 	Zone() string
 }
 
+var (
+	zone    string
+	zoneMtx sync.Mutex
+)
+
+func Zone() string {
+	zoneMtx.Lock()
+	defer zoneMtx.Unlock()
+
+	if zone == "" {
+		z, err := metadata.Zone()
+		if err != nil {
+			panic(err)
+		}
+		zone = z
+	}
+
+	return zone
+}
+
 var (
 	IsDevAppServer = false
 	IsFlex         = appengine.IsFlex
 	IsSecondGen    = appengine.IsSecondGen
 	IsStandard     = appengine.IsStandard
 )
+
+// Don't call this.  It exists to make NewAppengineInfoFromContext mockable
+func InternalNewAppengineInfoFromContext(c context.Context) AppengineInfo {
+	// if running in K8s, the following environment variable will always be set
+	if os.Getenv("KUBERNETES_SERVICE_HOST") != "" {
```

Comment:
I think I'll move this out into "InKubernetes()" or something; because I think I'll need this in the logging stuff in getLogger(), swapping out 'Type: "gae_app"'

Reply:
Just so logs get categorized in the correct place